Essentials/src/com/earth2me/essentials/commands/Commandnear.java

```diff
@@ -96,7 +97,15 @@ private String getLocal(final Server server, final User user, final long radius)
 
                 final long delta = (long) playerLoc.distanceSquared(loc);
                 if (delta < radiusSquared) {
-                    nearbyPlayers.offer(player);
+                    if (!player.isAuthorized("essentials.near.exempt")) {
```

Comment:
These checks can be and thus should be far higher up before we even grab their location (even in the if on line 92).

Its a minor optimization, but this mixes different checks in different places.